### ISSUE TITLE
"no-undef" does not catch undefined variables used with typeof

### ISSUE THREAD

**Issue by danielrentz** [eslint/eslint#9631]
**Tell us about your environment**

* **ESLint Version:**
4.6.1 (via grunt-eslint 20.1.0)
* **Node Version:**
v6.10.3
* **npm Version:**
v3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    rules: {
        'no-undef': 'error'
   }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function f1() {
    return foo;
}
function f2() {
    return typeof foo;
}
```

```bash

```

**What did you expect to happen?**
ESLint catches both undefined "foo"s.

**What actually happened? Please include the actual, raw output from ESLint.**
ESLint catches first "foo" only, but not the second.

**Comment by mysticatea** [comment]
Thank you for this issue.

This is intentional behavior because `typeof` operator can check undefined variables by the language spec.
The `no-undef` rule has an option about `typedef`: https://eslint.org/docs/rules/no-undef#options

**Comment by danielrentz** [comment]
Thank you for the quick answer, indeed I missed that nice option :)
